glfw/src/Window.zig

```diff
@@ -205,66 +203,124 @@ pub const Hint = enum(c_int) {
     x11_instance_name = c.GLFW_X11_INSTANCE_NAME,
 };
 
-/// Sets the specified window hint to the desired value.
-///
-/// This function sets hints for the next call to glfw.Window.create. The hints, once set, retain
-/// their values until changed by a call to this function or glfw.window.defaultHints, or until the
-/// library is terminated.
-///
-/// This function does not check whether the specified hint values are valid. If you set hints to
-/// invalid values this will instead be reported by the next call to glfw.createWindow.
-///
-/// Some hints are platform specific. These may be set on any platform but they will only affect
-/// their specific platform. Other platforms will ignore them.
-///
-/// Possible errors include glfw.Error.NotInitialized and glfw.Error.InvalidEnum.
-///
-/// @pointer_lifetime in the event that value is of a str type, the specified string is copied before this function returns.
-///
-/// @thread_safety This function must only be called from the main thread.
-///
-/// see also: window_hints, glfw.Window.defaultHints
-pub inline fn hint(h: Hint, value: anytype) Error!void {
-    const value_type = @TypeOf(value);
-    const value_type_info: std.builtin.TypeInfo = @typeInfo(value_type);
-
-    switch (value_type_info) {
-        .Int, .ComptimeInt => {
-            c.glfwWindowHint(@enumToInt(h), @intCast(c_int, value));
-        },
-        .Bool => {
-            const int_value = @boolToInt(value);
-            c.glfwWindowHint(@enumToInt(h), @intCast(c_int, int_value));
-        },
-        .Enum => {
-            const int_value = @enumToInt(value);
-            c.glfwWindowHint(@enumToInt(h), @intCast(c_int, int_value));
-        },
-        .Array => |arr_type| {
-            if (arr_type.child != u8) {
-                @compileError("expected array of u8, got " ++ @typeName(arr_type));
-            }
-            c.glfwWindowHintString(@enumToInt(h), &value[0]);
-        },
-        .Pointer => |pointer_info| {
-            const pointed_type = @typeInfo(pointer_info.child);
-            switch (pointed_type) {
-                .Array => |arr_type| {
-                    if (arr_type.child != u8) {
-                        @compileError("expected pointer to array of u8, got " ++ @typeName(arr_type));
-                    }
-                },
-                else => @compileError("expected pointer to array, got " ++ @typeName(pointed_type)),
+/// Window hints
+pub const Hints = struct {
```

Comment:
Because these are the only public hints exposed by the API, I think that we need to move all the docstrings from the current `const Hint = enum(c_int) {` into the fields here. Otherwise, once a Zig documentation generator exists (or even if you're just looking at this code for usage) you would not find any docstrings on these fields.

Reply:
Here, I'm actually unsure if all of the comments in `Hint` are totally necessary. They about double the length of the struct's declaration, and don't (in my opinion) add that much information. All of them follow the pattern:
```zig
    /// Window {name} hint
    {name},
```
Save for a few, like `center_cursor` and `focus_on_show`, and the framebuffer bits hints, which follow their own pattern.
But if you feel strongly about it, I can add the changes.

Reply:
Makes sense to me! I agree having comments just for the sake of it is bad practice, we should remove all such instances. They probably just ended up there as I didn't notice it. I should've checked the comments were actually useful before asking you to do this, sorry 😛

Reply:
All's good. I'll sort through them for a bit, and grab the ones that are meaningful, and polish them up.

Reply:
I've done the sorting; give it a quick look and see if there's anything amiss. If that looks good, then I feel confident about saying this is locked and loaded. I'll look at making a PR for that test/those tests concerning hints/attributes at some point in the near-ish future.